modules/standard/Data/Promise.enc

```diff
@@ -0,0 +1,32 @@
+-- Promises are futures that you can fulfil by explictly calling a method that you pass a value.
+module Promise
+
+-- Create a promise, together with a future that can be
+-- fulfilled using the promise:
+-- Once you call fulfil(..) on the promise object, the
+-- future returned by this call will be fulfilled.
+fun mkPromise[t]() : (Fut[t], Promise[t])
+  val f = EMBED (Fut[t]) future_mk(_ctx, _enc__type_t); END
+  (f, new Promise(f))
+end
+
+-- This should be linear, but we're hitting issue #770 then
+-- in the mean time, do not alias promises!
+unsafe class Promise[t]
```

Comment:
I wonder whether this can be refactored into two traits, one for the Fut and one linear one for handling Fulfilling. Then you could have `class Promise : Fut + linear Fulfiller` (or whatever) and can use `new Promise` to create instances, rather than the somewhat strange `mkPromise`.